### PR TITLE
refactor(process_tracker): attempt to identify unintended 5xx in process_tracker

### PR DESCRIPTION
**Pull request by Abhicodes-crypto**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR address possible unintended 5xx raised in process tracker flows. If not found 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

compiles

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/refunds.rs [hunk 1]
```diff
@@ -945,7 +945,7 @@ pub async fn add_refund_sync_task(
     };
 
     let response = db
-        .insert_process(process_tracker_entry)
+        .insert_process(process_tracker_entry) // if the process already exists in process tracker, i.e if it is is_db_unique_voilation should the error be ignored, this already happens in save cards flow
```
Comment (NishantJoshi00) [inline crates/router/src/core/refunds.rs:948]: please resolve the issue raised by `Github action`

**Review by NishantJoshi00 — COMMENTED** [review]
Other than that, looks good to me

**Review by NishantJoshi00 — COMMENTED** [review]
Other than that, looks good to me

File: crates/router/src/core/refunds.rs [hunk 2]
```diff
@@ -929,7 +929,10 @@ pub async fn add_refund_sync_task(
         .attach_printable_lazy(|| format!("unable to convert into value {:?}", &refund))?;
     let task = "SYNC_REFUND";
     let process_tracker_entry = storage::ProcessTrackerNew {
-        id: format!("{}_{}_{}", runner, task, refund.id),
+        id: format!(
+            "{}_{}_{}_{}",
+            runner, task, refund.merchant_id, refund.refund_id
```
Comment (jarnura) [inline crates/router/src/core/refunds.rs:934]: Change this to InternalRequestId

**Review by jarnura — APPROVED** [review]

File: crates/router/src/core/refunds.rs [hunk 3]
```diff
@@ -970,7 +973,10 @@ pub async fn add_refund_execute_task(
         .change_context(errors::ApiErrorResponse::InternalServerError)
         .attach_printable_lazy(|| format!("unable to convert into value {:?}", &refund))?;
     let process_tracker_entry = storage::ProcessTrackerNew {
-        id: format!("{}_{}_{}", runner, task, refund.id),
+        id: format!(
+            "{}_{}_{}_{}",
+            runner, task, refund.merchant_id, refund.refund_id
```
Comment (jarnura) [inline crates/router/src/core/refunds.rs:978]: Change this to InternalRequestId

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

**Pull request merged**